stl/inc/atomic

```diff
@@ -138,18 +138,17 @@ extern "C" inline void _Check_memory_order(const unsigned int _Order) noexcept {
     case _Atomic_memory_order_relaxed:                        \
         _Result = __iso_volatile_load##_Width(_Ptr);          \
         break;                                                \
+    case _Atomic_memory_order_release:                        \
+    case _Atomic_memory_order_acq_rel:                        \
+    default:                                                  \
+        _INVALID_MEMORY_ORDER;                                \
+        _FALLTHROUGH;                                         \

```

Comment:
Can be structured as:
```C++
if (_Order_var == _Atomic_memory_order_release || _Order_var == _Atomic_memory_order_acq_rel) {
     _INVALID_MEMORY_ORDER;
}

if (_Order_var == _Atomic_memory_order_relaxed) {
     _Result = __iso_volatile_load##_Width(_Ptr);
} else {
     _Result = __LOAD_ACQUIRE_ARM64(_Width, _Ptr);         \
     _Compiler_barrier();    
}
```

This avoids falling through, and likely to improve codegen in some cases, as #2110.
Occurs below.

Reply:
I am moderately persuaded by fallthrough being somewhat gross, weakly persuaded by possible codegen improvements (I don't have good intuition or ability to measure that), but strongly persuaded by `_Check_load_memory_order` already having centralized the ability to distinguish good kitties :smile_cat: from bad kitties :smirk_cat:. I'll audit the header, thanks.

Reply:
I got rid of as many `_FALLTHROUGH`s as I felt was reasonable here (the remaining ones seemed justified). Completely eliminating the abominable `_ATOMIC_STORE_PREFIX` made me very happy, thanks.